fraud/testing.go

```diff
@@ -0,0 +1,90 @@
+package fraud
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-blockservice"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/ipld"
+)
+
+type DummyService struct {
+}
+
+func (d *DummyService) Start(context.Context) error {
+	return nil
+}
+
+func (d *DummyService) Stop(context.Context) error {
+	return nil
+}
+
+func (d *DummyService) Broadcast(context.Context, Proof) error {
+	return nil
+}
+
+func (d *DummyService) Subscribe(ProofType) (Subscription, error) {
+	return &dummySubscription{}, nil
+}
+
+func (d *DummyService) RegisterUnmarshaler(ProofType, ProofUnmarshaler) error {
+	return nil
+}
+
+func (d *DummyService) UnregisterUnmarshaler(ProofType) error {
+	return nil
+}
+
+type dummySubscription struct {
+}
+
+func (d *dummySubscription) Proof(ctx context.Context) (Proof, error) {
+	<-ctx.Done()
+	return nil, ctx.Err()
+}
+
+func (d *dummySubscription) Cancel() {}
+
+type mockStore struct {
```

Comment:
we also have mockStore in `exchange_test`, maybe we can extract into its own testing file? This can be done in a later PR, otherwise we have duplicate code here

Reply:
Can you make an issue for this? and link to this PR

Reply:
https://github.com/celestiaorg/celestia-node/issues/853

Reply:
ack